### ISSUE TITLE
Netherlands countrywide imports broken.

### ISSUE THREAD

**Issue by berkes** [openaddresses/openaddresses#5019]
In PR #4573 used to fix #4572, the URL to get the dutch BAG database was moved. 

The download link, introduced in #4573 now redirects to a paid, commercial service. This makes the service break:

```
2020-02-15 09:13:26,847    DEBUG: URL says ".zip" for https://data.nlextract.nl/bag/csv/bag-adressen-laatst.csv.zip
2020-02-15 09:13:26,847    DEBUG: Guessed addresses-primary-82bbbc76.zip for https://data.nlextract.nl/bag/csv/bag-adressen-laatst.csv.zip
2020-02-15 09:13:26,847    DEBUG: Requesting https://data.nlextract.nl/bag/csv/bag-adressen-laatst.csv.zip with args None
2020-02-15 09:13:27,914     INFO: Downloaded 8349 bytes for file /tmp/worker-wpuoi423/work-5m7qpa2d/out/process_one-1e36r7ag/cache-b_8rfa3o/http/addresses-primary-82bbbc76.zip
2020-02-15 09:13:27,915     INFO: Cached data in file:///tmp/worker-wpuoi423/work-5m7qpa2d/out/process_one-1e36r7ag/cached/addresses-primary-82bbbc76.zip
2020-02-15 09:13:27,915    DEBUG: URL says ".zip" for file:///tmp/worker-wpuoi423/work-5m7qpa2d/out/process_one-1e36r7ag/cached/addresses-primary-82bbbc76.zip
2020-02-15 09:13:27,915    DEBUG: Guessed addresses-primary-e6b9e8bd.zip for file:///tmp/worker-wpuoi423/work-5m7qpa2d/out/process_one-1e36r7ag/cached/addresses-primary-82bbbc76.zip
2020-02-15 09:13:27,915    DEBUG: File exists /tmp/worker-wpuoi423/work-5m7qpa2d/out/process_one-1e36r7ag/conform-plivlg7_/http/addresses-primary-e6b9e8bd.zip
2020-02-15 09:13:27,916     INFO: Downloaded to ['/tmp/worker-wpuoi423/work-5m7qpa2d/out/process_one-1e36r7ag/conform-plivlg7_/http/addresses-primary-e6b9e8bd.zip']
2020-02-15 09:13:27,916  WARNING: Error in process_one.process()
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/openaddr/process_one.py", line 148, in process
    conform_result = conform(layer + '-' + data_source['name'], data_source, temp_dir, cache_result.todict())
  File "/usr/local/lib/python3.5/dist-packages/openaddr/__init__.py", line 155, in conform
    decompressed_paths = task2.decompress(downloaded_path, workdir, names)
  File "/usr/local/lib/python3.5/dist-packages/openaddr/conform.py", line 243, in decompress
    with ZipFile(source_path, 'r') as z:
  File "/usr/lib/python3.5/zipfile.py", line 1026, in __init__
    self._RealGetContents()
  File "/usr/lib/python3.5/zipfile.py", line 1093, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
```

The official dutch data-provider does not provide this data (**edit** as CSV). But as a ~nightly~ Monthly .zip file.

http://geodata.nationaalgeoregister.nl/inspireadressen/extract/inspireadressen.zip [warning: 1,7G]

This is all XML and rather convoluted. I'm not familiar  with the openaddress DSL nor its machinery to create a config to parse and extract that XML, though.

**Comment by berkes** [comment]
Ping to @stvno

**Comment by berkes** [comment]
The ATOM file is here: https://geodata.nationaalgeoregister.nl/inspireadressen/atom/inspireadressen.xml

It contains the URLS for the XML that describes content, structure, licence and links to other data.

**Comment by WouterVisscher** [comment]
@berkes for reference(/background information) the data provided though ```http://geodata.nationaalgeoregister.nl/inspireadressen/extract/inspireadressen.zip``` is the official data provided by the LV-BAG (the official dataset owner/data-provider) through the Dutch NSDI (national spatial data infrastructure) **PDOK**, that uses (for now) the URL https://geodata.nationaalgeoregister.nl... (dutch for "spatial data" and "national key registery")

This data set is the RAW data (official datamodel) and is updated every month. So the dutch data-provider **does** provide this data. The older 'source' NLextract id(/was) a initiative based on this data. In some other issue I see references too ESRI sources... those wil likely also be based on the same original data...

**Comment by berkes** [comment]
@WouterVisscher do you know who generates the version on s3? Someone is generating them, probably with NLExtract and then publishes them on s3. if the Kadaster itself publishes it there, that would make it the best resource by far. But if it is a volunteer, it would at least be crucial to know who does this so that we can reach out in case of errors.

**Comment by WouterVisscher** [comment]
@berkes 
1. do you know who generates the version on s3? **nope**
2. probably with NLExtract? **nope**, this is the 'official LV-BAG data from (april 2014...) so no NLExtract or other mutations/enrichment/or so on...
3. if the Kadaster itself publishes it there, that would make it the best resource by far: **nope** We (Kadaster/PDOK) don't have to publish it there, we already publish through https://geodata.nationaalgeoregister.nl/inspireadressen/extract/inspireadressen.zip (and that like is updated every month)
4. But if it is a volunteer: I presume that someone download it in 2014 from us (PDOK) and uploaded it to s3...

**Comment by berkes** [comment]
@WouterVisscher thanks for the reply. We were talking about two different s3 uploads, apparently. Sorry for the confusion. I'll edit the issue at the top to reflect the current situation better.

The JS script you point to, no longer works. I tried running it with the most recent inspireaddressen.zip as advertised in the ATOM file (the url geodata.... you point to). Problems with that conversion are i) dependencies that no longer work/run and ii) hardcoded assumptions about the datamodel that changed ever so slightly. 

I concluded that this JS file is therefore unmaintained, and unused for a long time now.

The version that was used by OA is a CSV. So, OA imports a dutch CSV, changes it a little and then re-publishes a countrywide.csv from that.

The CSV that OA imported, was provided by nlextract.nl. Untill february, that CSV could be downloaded for free. Now it is behind a paywall.

This CSV is generated with NLExtract from inspireaddressen.zip. The process to do so is rather complex and requires an intermediate postgis database. OA cannot run such a process. OA can run (simple) python or js to convert XML into CSV, but not with an intermediate postgis database. I'm not aware of (working and maintained) tooling that can convert the zipped sets of XML into CSV without such intermediate storage.

**Comment by WouterVisscher** [comment]
aha! that makes more sense 😄 
A shame that that code didn't work anymore...

Yeah I was taking about the: ```http://gis-data.s3.amazonaws.com/nl-gov/inspireadressen.zip```
not the ```https://data.nlextract.nl/bag/csv/bag-adressen-laatst.csv.zip``` sorry for the confusion.

Regarding the Dutch addresses as CSV (and the NLextract CSV now behind a paywall), we (Kadaster/PDOK) do get more requests for a 'data format' like that... Problem is that we (PDOK) are not the once to decide that in what formats the data is provided (contrary to popular belief...) I would like to make the suggestion that it would be 'good' to let the LV-BAG (the once that do decide) know that a CSV extract of the address is a useful product (or if they are 'aware' of this problem, or maybe have a alternative...). [contact info](https://www.kadaster.nl/zakelijk/over-ons/klantenservice-zakelijk) [contact form](https://formulieren.kadaster.nl/contact_bag)

**Comment by arch0345** [comment]
Resolved in #6365
